### PR TITLE
Make Duration/Period factories inherit AbstractArgumentFactory

### PR DESCRIPTION
**Pull request by arteam**
Doing so, we can avoid additional checks for confirming factory to the right type and make the factories more generic-friendly.

References #686

### CODE REVIEW

File: postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java [hunk 1]
```diff
@@ -42,14 +40,16 @@
...
+        super(Types.OTHER);
+    }
+
     @Override
-    public Optional<Argument> build(Type type, Object value, ConfigRegistry config) {
-        if (Duration.class != type) {
-            return Optional.empty();
-        }
+    public Argument build( Duration value, ConfigRegistry config) {
         if (null == value) {
```
Comment (stevenschlansker) [inline postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java]: Are these null checks now redundant?

**Comment by stevenschlansker** [comment]
Looking good, although @qualidafial it looks like the oracle bit broke again in Travis?

File: postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java (hunk 1 above)
Comment (qualidafial) [inline postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java]: Yep!

**Comment by qualidafial** [comment]
Merged from jdbi3, removed the redundant null check--build looks good. Merging. Woot!

**Comment by stevenschlansker** [comment]
cc @pennello

File: postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java [hunk 2]
```diff
@@ -42,14 +40,16 @@
...
+    public DurationArgumentFactory(){
+        super(Types.OTHER);
+    }
+
     @Override
-    public Optional<Argument> build(Type type, Object value, ConfigRegistry config) {
-        if (Duration.class != type) {
-            return Optional.empty();
-        }
+    public Argument build( Duration value, ConfigRegistry config) {
```
Comment (pennello) [inline postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java]: Extremely minor typo: space before `Duration`.

Comment (qualidafial) [inline postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java]: Should have used a tab :trollface:

Comment (pennello) [inline postgres/src/main/java/org/jdbi/v3/postgres/DurationArgumentFactory.java]: Ha!

**Pull request merged**